File: api/application/src/main/java/net/corda/v5/application/persistence/CordaPersistenceException.java
package net.corda.v5.application.persistence;

import net.corda.v5.base.annotations.ConstructorForDeserialization;
import net.corda.v5.base.exceptions.CordaRuntimeException;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/**
 * Exception that encapsulates errors during persistence.
 */
public final class CordaPersistenceException extends CordaRuntimeException {

    /**
     * @param message The exception message.
     * @param cause Optional throwable that was caught.
     */
    @ConstructorForDeserialization
    public CordaPersistenceException(@NotNull String message, @Nullable Throwable cause) {
        super(message, cause);
    }

    /**
     * Creates a {@link CordaPersistenceException} without a {@code cause}.
     *
     * @param message The exception message.
     */
    public CordaPersistenceException(@NotNull String message) {
        this(message, null);
    }
}

File: application/src/main/kotlin/net/corda/v5/application/persistence/CordaPersistenceException.kt
package net.corda.v5.application.persistence

import net.corda.v5.base.exceptions.CordaRuntimeException

/**
 * Exception that encapsulates errors during persistence.
 *
 * @param message The exception message.
 * @param cause Optional throwable that was caught.
 *
 * @constructor Creates a [CordaPersistenceException] with a [cause].
 */
class CordaPersistenceException(message: String, cause: Throwable?) : CordaRuntimeException(message, cause) {

    /**
     * Creates a [CordaPersistenceException] without a [cause].
     *
     * @param message The exception message.
     */
    constructor(message: String) : this(message, null)
}